File: 202-happy-number/happy-number.cpp
class Solution {
public:
    bool isHappy(int n) {
        set<string> s;
        string curr=to_string(n);
        int sum=0;
        while(!s.count(curr)){
            for(int i=0;i<curr.size();i++){
                sum+=(curr[i]-'0')*(curr[i]-'0');
            }
            if(sum==1) return true;
            s.insert(curr);
            curr=to_string(sum);
            sum=0;
            
        }
        return false;
    }
};

File: 202-happy-number/happy-number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        
            sum=0
            s=set()
            while sum!=1:
                sum=0
                while n:
                    sum+=(n%10)**2
                    n=n//10
                if sum in s:
                    return False
                s.add(sum)
                
                n=sum
            return True

